### DISCUSSION TITLE
Recover deleted files ?

### DISCUSSION THREAD

**Discussion by arnaudrichard** [jivanpal/apfs-tools#51] [Q&A]
Hello,
I'm a little bit confused about what drat can or can't do.
I would like to recover files I just deleted (moved to Trash, then emptied the trash) on a SATA SSD APFS volume.
Is it possible with drat ? How to search for them as I guess they are no longer listed in the B-tree ?
Thx!

**Comment by jivanpal** [comment]
"Undelete" relies on the ability to explore past APFS transactions, but Drat doesn't yet have great support for exploring past APFS transactions. Rather, it currently just tries to work with the latest intact transaction, which is usually suitable in a situation where the partition has become corrupted. Sorry I can't be of better help right now!
